api/agent/drivers/docker/docker_client.go

```diff
@@ -308,6 +311,19 @@ func (d *dockerWrap) LoadImages(ctx context.Context, filePath string) error {
 	})
 }
 
+func (d *dockerWrap) ListImages(ctx context.Context) (imgs []docker.APIImages, err error) {
+	ctx, closer := makeTracker(ctx, "list_docker_images")
+	defer closer()
+
+	logger := common.Logger(ctx).WithField("docker_cmd", "ListImages")
+	err = d.retry(ctx, logger, func() error {
+		imgs, err = d.docker.ListImages(docker.ListImagesOptions{All: false})
```

Comment:
most of the operations in this file allowing specifying the docker options as well (like this one). I forget how well articulated the intent of this package was but it was intended to mirror the parts of the fsouza API we need but just add retries, stats, etc. could you make the signature for the `dockerClient` interface `ListImages(docker.ListImagesOptions)` as well, please? I assume there's a context in the options somewhere or the fsouza client should have a `ListImageWithContext` method as well. we need to thread the context into the underlying operation, as well, to avoid leaking goroutines / connections / etc.